common/src/test/java/net/adoptopenjdk/icedteaweb/jvm/JvmUtilsTest.java

```diff
@@ -90,4 +96,114 @@ public void testJava8JavaVMArgs() {
+        }
+
+        final String java_vm_args = "-Dsun.java2d.d3d=false -Dsun.java2d.dpiaware=false --add-opens=java.desktop/sun.print=ALL-UNNAMED --add-exports=java.desktop/sun.print=ALL-UNNAMED --add-exports=java.desktop/sun.swing=ALL-UNNAMED --add-exports=java.desktop/sun.swing.table=ALL-UNNAMED --add-exports=java.desktop/sun.swing.plaf.synth=ALL-UNNAMED --add-opens=java.desktop/javax.swing.plaf.synth=ALL-UNNAMED --add-opens=java.desktop/javax.swing.plaf.basic=ALL-UNNAMED --add-opens=java.desktop/javax.swing=ALL-UNNAMED --add-opens=java.desktop/javax.swing.tree=ALL-UNNAMED --add-opens=java.desktop/java.awt.event=ALL-UNNAMED --add-exports=java.desktop/com.sun.java.swing.plaf.windows=ALL-UNNAMED --add-exports=java.desktop/sun.awt.shell=ALL-UNNAMED --add-exports=java.desktop/com.sun.awt=ALL-UNNAMED --add-exports=java.base/sun.security.action=ALL-UNNAMED";
+        try {
+            JvmUtils.checkVMArgs(java_vm_args);
+        } catch (IllegalArgumentException ile) {
+            fail(ile.getMessage());
+        }
+
+        final String java_vm_bad_args = "-Dsun.java2d.d3d=false -Dsun.java2d.dpiaware=false  --add-opens=java.desktop/sun.print=ALL-UNNAMED --list-modules --add-exports=java.desktop/sun.print=ALL-UNNAMED --add-exports=java.desktop/sun.swing=ALL-UNNAMED --add-exports=java.desktop/sun.swing.table=ALL-UNNAMED --add-exports=java.desktop/sun.swing.plaf.synth=ALL-UNNAMED --add-opens=java.desktop/javax.swing.plaf.synth=ALL-UNNAMED --add-opens=java.desktop/javax.swing.plaf.basic=ALL-UNNAMED --add-opens=java.desktop/javax.swing=ALL-UNNAMED --add-opens=java.desktop/javax.swing.tree=ALL-UNNAMED --add-opens=java.desktop/java.awt.event=ALL-UNNAMED --add-exports=java.desktop/com.sun.java.swing.plaf.windows=ALL-UNNAMED --add-exports=java.desktop/sun.awt.shell=ALL-UNNAMED --add-exports=java.desktop/com.sun.awt=ALL-UNNAMED --add-exports=java.base/sun.security.action=ALL-UNNAMED";
+        try {
+            JvmUtils.checkVMArgs(java_vm_bad_args);
+            fail("Illegal arg accepted!");
+        } catch (IllegalArgumentException ile) {
+            assertTrue(true);
+        }
+    }
+
+    @Test
+    public void testPredefModuleVMArgs() {
+
+        final List<String> result = JvmUtils.mergeJavaModulesVMArgs(Collections.emptyList());
+
+        assertEquals(JvmUtils.getPredefinedJavaModulesVMArgumentsMap().size(), result.size());
+        assertTrue(result.contains("--add-reads=java.base=ALL-UNNAMED,java.desktop"));
+        assertTrue(result.contains("--add-exports=java.desktop/sun.applet=ALL-UNNAMED,java.desktop,jdk.jsobject"));
+    }
+
+    @Test
+    public void testMergeJavaModuleVMArgs() {
+
+        final String[] usrStrArr = new String[] {"-DnoModuleArg=bbb", "--add-reads=java.base=ALL-UNNAMED,java.xxx"};
+        final List<String> usrDefArgs = new ArrayList(Arrays.asList(usrStrArr));
+
+        final List<String> result = JvmUtils.mergeJavaModulesVMArgs(usrDefArgs);
+        assertTrue(result.contains("-DnoModuleArg=bbb"));
```

Comment:
can you please add the following assertion:

`assertEquals(JvmUtils.getPredefinedJavaModulesVMArgumentsMap().size() + 1, result.size());`

Reply:
Right